ceph/rados/crushtool_workflows.py

```diff
@@ -640,3 +640,25 @@ def test_crush_map_bin(
             return False, " "
         log.debug(f"Successfully tested the bin file for {test} and output : {out}")
         return True, out
+
+    def add_crush_rule(self, rule_name: str, rules: str):
+        """
+        Method to add new crush rules into the cluster
+        Args:
+            rules: Crush rules.
+            rule_name: name of the rule
+        """
+        _, loc = self.generate_crush_map_bin()
+        _, loc = self.decompile_crush_map_txt(source_loc=loc)
+
+        # Adding the crush rules into the file
+        rule_str = f"rule {rule_name} {{ \n{rules}\n}}"
+        cmd = f"echo '{rule_str}' >> {loc}"
+        self.client.exec_command(cmd=cmd, sudo=True)
+
+        _, loc = self.compile_crush_map_txt(source_loc=loc)
+        if not self.set_crush_map_bin(loc=loc):
+            log.error(f"Failed to set rules {rules} on the cluster")
+            return False
+        log.info(f"Successfully set rules {rules} with name {rule_name} on the cluster")
```

Comment:
Just a hunch, but wouldn't this statement become very huge and disoriented if we include the `{rules}` here? Maybe just mentioning the rule_name is enough. WDYT?

Reply:
Yes, Sounds good. Removed the rules string.